src/overlay.js

```diff
@@ -105,29 +111,35 @@
             options.location.x,
             options.location.y,
             options.location.width,
-            options.location.height
-        );
-        this.position   = new $.Point(
-            options.location.x,
-            options.location.y
-        );
-        this.size       = new $.Point(
-            options.location.width,
-            options.location.height
-        );
-        this.style      = options.element.style;
-        // rects are always top-left
-        this.placement  = options.location instanceof $.Point ?
+            options.location.height);
+
+        // this.position is never read by this class but is kept for backward
+        // compatibility
+        this.position = this.bounds.getTopLeft();
+
+        // this.size is only used by PointOverlay with options.checkResize === false
+        this.size = this.bounds.getSize();
+
+        this.style = options.element.style;
+
+        // rects are always top-left (RectOverlays don't use placement)
+        this.placement = options.location instanceof $.Point ?
             options.placement : $.Placement.TOP_LEFT;
         this.onDraw = options.onDraw;
         this.checkResize = options.checkResize === undefined ?
             true : options.checkResize;
+        this.scaleWidth = options.scaleWidth === undefined ?
+            true : options.scaleWidth;
+        this.scaleHeight = options.scaleHeight === undefined ?
+            true : options.scaleHeight;
     };
 
     /** @lends OpenSeadragon.Overlay.prototype */
     $.Overlay.prototype = {
 
         /**
+         * Internal function to adjust the position of a PointOverlay
```

Comment:
If we don't literally have an object called `PointOverlay` we probably shouldn't refer to such a thing. Maybe call it a "point Overlay" or "point-based Overlay"?